### DISCUSSION TITLE
Smuggler data export doesn't work

### DISCUSSION THREAD

**Discussion by nlhieke** [ravendb/ravendb#16415] [Q&A] [answered]
When I try to create a data dump (export) of a database (about 500 MB) the method WaitForCompletionAsync hangs forever.
I am doing the same as mentioned in the example on:
https://ravendb.net/docs/article-page/5.4/csharp/client-api/smuggler/what-is-smuggler#example
Also no progress is reported if I add an event handler. I have tried the obvious like different database, checking file rights, etc. 
Creating a backup (ravendbdump) from Studio succeeds within a couple of seconds.
We are running on premise. I have tried RavenDB Client 5.4.4 and 5.4.103. Database Server and Studio version is 5.3.102.

**Comment by ayende** [comment]
Can you share the full code?

**Reply by nlhieke** [reply]
```
var exportOperation = await store
    .Smuggler
    .ExportAsync(
        new DatabaseSmugglerExportOptions
        {
            OperateOnTypes = DatabaseItemType.Documents
        },
        @"C:\database.ravendbdump");

await exportOperation.WaitForCompletionAsync();
```

**Reply by ayende** [reply]
You are likely trying to write to a location that you don't have permissions to.
Can you try: `        @".\database.ravendbdump");` Instead, or something similar?

**Reply by nlhieke** [reply]
I already tried it (and once again, just yet). I am working on my own pc, on which I have admin rights. The database is running on a server.

**Reply by ayende** [reply]
If you run in the debugger, with stop on exception, do you get anything?

**Reply by nlhieke** [reply]
I am running in the debugger and stepping through the code. Stepping over WaitForCompletionAsync results in a hang.

**Reply by ayende** [reply]
what happens if you enable stop on exceptions?

**Reply by nlhieke** [reply]
Same result when all exception checkboxes are checked...

**Reply by ayende** [reply]
Are you available for a quick call?

**Reply by nlhieke** [reply]
I was busy, but I am available now.

**Comment by garayx** [comment]
I wonder if the clients DocumentStore is connected to the server ? Can you try to store or load a document from server? does it work?

**Reply by nlhieke** [reply]
What do you mean exactly? Can you do that directly on the store, without creating a session?

**Reply by garayx** [reply]
I meant [open a session](https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/opening-a-session#open-session-example), and then use [session.Load](https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/loading-entities#example) or [session.Store](https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/storing-entities#example), does it work?

**Reply by nlhieke** [reply]
Yes that works. I am using it for a while now, but I recently wanted to use export functionality which doesn't work.

**Reply by grisha-kotler** [reply]
Can you try using the lastest 5.4.103 (server)?

**Reply by nlhieke** [reply, marked as answer]
Thanks! That does the trick! Very strange though, that is doesn't work on 5.3.102.
Another question: The documentation states that the export is in [json format](https://ravendb.net/docs/article-page/5.4/csharp/client-api/smuggler/what-is-smuggler#:~:text=a%20database%20using-,JSON,-format.%0AIt) but is some sort of binary format...

**Reply by grisha-kotler** [reply]
It's a `GZip` archive. Inside you'll find the JSON.

**Reply by nlhieke** [reply]
Thanks again! It all works now!
